src/streaming/utils/RequestModifier.js

```diff
@@ -31,20 +31,43 @@
 
 import FactoryMaker from '../../core/FactoryMaker';
 
+export function modifyRequest(httpRequest, requestModifier) {
+    if (!(requestModifier && requestModifier.modifyRequest)) {
```

Comment:
There is a small performance and timing impact here by always calling modifyRequest. This gaurantees the rest of the code will occur on nextTick even when no requestModifier is supplied. Do you think it would be better to avoid this impact or are you ok with this slight change?

Reply:
The code has been updated to only apply the async modifier if it exists.